components/infobox/extensions/wikis/rocketleague/years_active.lua

```diff
@@ -7,20 +7,94 @@
 --
 
 local Class = require('Module:Class')
+local Lpdb = require('Module:Lpdb')
 local Lua = require('Module:Lua')
+local Set = require('Module:Set')
+local Table = require('Module:Table')
 
 local CustomActiveYears = Lua.import('Module:YearsActive/Base', {requireDevIfEnabled = true})
 
 -- wiki specific settings
 CustomActiveYears.defaultNumberOfStoredPlayersPerPlacement = 6
 CustomActiveYears.additionalConditions = ''
 
+local _TALENT_POSITIONS = {
+	'Analyst',
+	'Caster',
+	'Caster/Analyst',
+	'Commentator',
+	'Commentator/Analyst',
+	'Desk Host',
+	'Host',
+	'Reporter',
+	'Stage Host',
+}
+
+local _OBSERVER_POSITIONS = {
+	'Observer',
+	'Observer/Producer',
+	'Producer/Observer',
+}
+
 -- legacy entry point
 function CustomActiveYears.get(input)
 	-- if invoked directly input == args
 	-- if passed from modules it might be a table that holds the args table
 	local args = input.args or input
-	return CustomActiveYears.display(args)
+	local display = CustomActiveYears.display(args)
+	return display ~= 'Player has no results.' and display or nil
+end
+
+function CustomActiveYears.getTalent(talent)
+	return CustomActiveYears._getBroadcaster(
+		CustomActiveYears._getBroadcastConditions(talent, _TALENT_POSITIONS)
+	)
+end
+
+function CustomActiveYears.getObserver(observer)
+	return CustomActiveYears._getBroadcaster(
+		CustomActiveYears._getBroadcastConditions(observer, _OBSERVER_POSITIONS)
+	)
+end
```

Comment:
Unnecessary split imo, comes with a too big maintenance load (there will be a continuous need to update).
The other tables (achievement, results) already have them shared

Reply:
I see talent and observer/producer as different roles (I assume that's what you mean by calling it a split), and we have observers with pages so I think it is fitting to show their years active in their infobox. We feature observers on every single event's broadcaster table, so I think it just makes sense to include them.

Reply:
Or do you mean I should query LPDB only once and calculate it from the one result?